File: packages/sdk/src/resources/skills.ts
import type { HttpClient } from "../http.js";
import type { SkillDetail, SkillSummary } from "../types.js";

export class SkillsResource {
  constructor(private readonly http: HttpClient) {}

  async list(options: { category?: string } = {}): Promise<SkillSummary[]> {
    const resp = await this.http.request<{ skills: SkillSummary[]; total: number }>({
      method: "GET",
      path: "/api/v3/mcp/skills",
      query: { category: options.category },
    });
    return resp.skills;
  }

  get(skillId: string): Promise<SkillDetail> {
    return this.http.request<SkillDetail>({
      method: "GET",
      path: `/api/v3/mcp/skills/${encodeURIComponent(skillId)}`,
    });
  }
}

File: python/src/apiz/_resources/skills.py
from __future__ import annotations

from typing import Optional
from urllib.parse import quote

from .._http import AsyncHttpClient, SyncHttpClient
from .._types import SkillDetail, SkillSummary


class SkillsResource:
    def __init__(self, http: SyncHttpClient) -> None:
        self._http = http

    def list(self, *, category: Optional[str] = None) -> list[SkillSummary]:
        data = self._http.request(
            "GET", "/api/v3/mcp/skills", query={"category": category}
        )
        items = (data or {}).get("skills", [])
        return [SkillSummary.model_validate(s) for s in items]

    def get(self, skill_id: str) -> SkillDetail:
        data = self._http.request(
            "GET", f"/api/v3/mcp/skills/{quote(skill_id, safe='')}"
        )
        return SkillDetail.model_validate(data)


class AsyncSkillsResource:
    def __init__(self, http: AsyncHttpClient) -> None:
        self._http = http

    async def list(self, *, category: Optional[str] = None) -> list[SkillSummary]:
        data = await self._http.request(
            "GET", "/api/v3/mcp/skills", query={"category": category}
        )
        items = (data or {}).get("skills", [])
        return [SkillSummary.model_validate(s) for s in items]

    async def get(self, skill_id: str) -> SkillDetail:
        data = await self._http.request(
            "GET", f"/api/v3/mcp/skills/{quote(skill_id, safe='')}"
        )
        return SkillDetail.model_validate(data)
